backend/knowledge_seeding_system.py

```
@@ -27,15 +27,74 @@
+    
+    def __init__(self, connection_string: str):
+        self.connection_string = connection_string
+        self._connection = None
+    
+    async def acquire(self):
+        """Return an async connection compatible with asyncpg API"""
+        if PSYCOPG_AVAILABLE:
+            conn = await AsyncConnection.connect(self.connection_string, row_factory=dict_row)
+            return AsyncPGCompatConnection(conn)
+        else:
+            raise ImportError("psycopg not available")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)